lib/_http_agent.js

```diff
@@ -392,13 +413,27 @@ Agent.prototype.removeSocket = function removeSocket(s, options) {
         // Don't leak
         if (sockets[name].length === 0)
           delete sockets[name];
+        this.totalSocketCount--;
       }
     }
   }
 
+  let req;
   if (this.requests[name] && this.requests[name].length) {
     debug('removeSocket, have a request, make a socket');
-    const req = this.requests[name][0];
+    req = this.requests[name][0];
+  } else {
```

Comment:
What's the reason for this newly added block, it doesn't seem to have anything to do with `maxTotalSockets` ?

Reply:
Because of different origin, pending requests cannot be resolved.

Reply:
eg:

Set conditions:
```
maxTotalSockets: 3

'localhost:3500:': Make 3 requests
'localhost:3600:': Make 3 requests
```

Socket allocation for two origin：
```
'localhost:3500:': 3
'localhost:3600:': 0
```
In the above case, requests from `'localhost:3600:'`  will not be resolved.